spring-rabbit/src/main/java/org/springframework/amqp/rabbit/config/ListenerContainerFactoryBean.java

```diff
@@ -211,7 +212,7 @@ public void setExposeListenerChannel(boolean exposeListenerChannel) {
 		this.exposeListenerChannel = exposeListenerChannel;
 	}
 
-	public void setMessageListener(Object messageListener) {
+	public void setMessageListener(MessageListener messageListener) {
```

Comment:
This is going to be a bit of breaking change, but I'm fine with the fix as far as the XML parser is not confused with two different setters.